controllers/refresolver/istiovirtualservice/istiovirtualservice.go

```diff
@@ -0,0 +1,219 @@
+package istiovirtualservice
+
+/*
+Copyright 2022 The k8gb Contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Generated by GoLic, for more details see: https://github.com/AbsaOSS/golic
+*/
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	k8gbv1beta1 "github.com/k8gb-io/k8gb/api/v1beta1"
+	"github.com/k8gb-io/k8gb/controllers/logging"
+	"github.com/k8gb-io/k8gb/controllers/utils"
+	istio "istio.io/client-go/pkg/apis/networking/v1beta1"
```

Comment:
Related to the proposal of the resourceRef spec modification - how we handle the future upgrade situation of `v1beta1` to `v1` in this case? and maybe supporting both versions in parallel for the time being?

Reply:
I didn't realize there was a more recent version of the API 😅
Yes, supporting multiple versions would be necessary to support upgrades. We can do that.

In that case would you "duplicate" also the tests? It will be mostly copy paste and it will be make our pipelines slower. But it is probably the only way we have to guarantee both versions work.

Could we

Reply:
The `v1` golang library is compatible with `v1beta1` and `v1` CRDs, at least for the fields that we are interested it.
Which makes sense, because the istio controller should be able to handle `apiVersion` upgrades. That means we don't have to have an implementation of the reference resolver for each `apiVersion`, we can rely on `v1` for both.

That might not be the case in the future though, at some point apiVersions will be deprecated. For that I created a workflow that replaces `networking.k8s.io/v1` with `networking.k8s.io/v1beta1` and is only triggered when the `istio` tag is added to the PR.